File: VB/Scaffolding.CustomFilter/IssueContextDataModel/IIssueContextUnitOfWork.vb
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.CustomFilter.Common.Utils
Imports Scaffolding.CustomFilter.Common.DataModel
Imports Scaffolding.CustomFilter.Common.DataModel.EntityFramework
Imports Scaffolding.CustomFilter.Model

Namespace Scaffolding.CustomFilter.IssueContextDataModel
    ''' <summary>
    ''' IIssueContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    ''' </summary>
    Public Interface IIssueContextUnitOfWork
        Inherits IUnitOfWork

        ''' <summary>
        ''' The Issue entities repository.
        ''' </summary>
        ReadOnly Property Issues() As IRepository(Of Issue, Integer)
    End Interface
End Namespace

File: CS/Scaffolding.CustomFilter/IssueContextDataModel/IIssueContextUnitOfWork.cs
using DevExpress.Mvvm.DataModel;
using Scaffolding.CustomFilter.Model;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Scaffolding.CustomFilter.IssueContextDataModel {

    /// <summary>
    /// IIssueContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    /// </summary>
    public interface IIssueContextUnitOfWork : IUnitOfWork {

        /// <summary>
        /// The Issue entities repository.
        /// </summary>
        IRepository<Issue, int> Issues { get; }
    }
}
